packages/backend-common/src/database/connectors/postgres.ts

```diff
@@ -35,6 +35,14 @@ export function createPgDatabaseClient(
 ) {
   const knexConfig = buildPgDatabaseConfig(dbConfig, overrides);
   const database = knexFactory(knexConfig);
+
+  const owner = dbConfig.getOptionalString('setOwner');
+
+  if (owner) {
+    database.client.pool.on('createSuccess', (_event: any, pgClient: any) => {
+      pgClient.query(`SET ROLE ${owner}`, () => {});
```

Comment:
Can we do `await client.raw(...)` instead of the empty callback?

Reply:
No, the client is not a Knex Object, but a 'raw' connection, which in this case is a [pg.client](https://node-postgres.com/apis/client#clientquery). 

I fixed the typing and added a callback so it doesn't fail silently.

Reply:
https://node-postgres.com/apis/client#promise-api Looks like there's a promise interface though. So you should be able to do something like this:

```ts
  if (role) {
    database.client.pool.on(
      'createSuccess',
      async (_: number, pgClient: Client) => {
        await pgClient.query(`SET ROLE ${role}`);
      },
    );
  }
  ```